src/decorations/prepareIcon.ts

```diff
@@ -0,0 +1,32 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import { ExtensionContext } from 'vscode';
+const ICONS_PATH = path.join('out', 'icons');
+
+export default function prepareIcon(
+  context: ExtensionContext,
+  state: string,
+  source: string,
+  color?: string
+): string {
+  const resolvePath = (...args: string[]): string => {
+    return context.asAbsolutePath(path.join(...args));
+  };
+
+  const resultIconPath = resolvePath(ICONS_PATH, `${state}.svg`);
+  let result = source.toString();
+
+  if (color !== undefined) {
+    result = result.replace('fill="currentColor"', `fill="${color}"`);
+  }
+
+  if (!fs.existsSync(resultIconPath) || fs.readFileSync(resultIconPath).toString() !== result) {
```

Comment:
It's invoked only in constructors, not on every update.

Reply:
if you are relying on the caller to cache the result otherwise suffer performance hit, then we should at least document it. IMHO, I think it might be more robust if this method itself does the caching so it doesn't have to rely on every caller.

Reply:
Cache will only in incentivise calling this over and over, while adding complexity and being fairly hard to test. 

For now I renamed this to `prepareIconFile`. **This is not a `getIcon` function.** Please let me know if comments provided in code I will push next are up to your standards.

Alternatively we could do something like below, but I have no good idea how on how to test it. Export? With `rewire` package?
``` ts
const cache: Record<string, string> = {}
... // in scope:
    const cacheKey = `${iconName}/${source}/${color || ''}`
    if (cache[cacheKey]) return cache[cacheKey]
... // || update cache later
```

Reply:
I think this PR is already bigger than I originally anticipated, so yes I think we can leave it with a clear comment for now.